storage/src/main/scala/coop/rchain/storage/IStore.scala

```diff
@@ -37,15 +37,15 @@ trait IStore[C, P, A, K] {
 
   def putK(txn: T, channels: List[C], patterns: List[P], k: K): Unit
 
-  def getPs(txn: T, channels: List[C]): List[P]
+  def getPs(txn: T, channels: List[C]): List[List[P]]
 
   def getAs(txn: T, channels: List[C]): List[A]
 
-  def getK(txn: T, curr: List[C]): Option[(List[P], K)]
+  def getPsK(txn: T, curr: List[C]): List[(List[P], K)]
 
-  def removeA(txn: T, channels: List[C], index: Int): Unit
+  def removeA(txn: T, channel: C, index: Int): Unit
 
-  def removeK(txn: T, channels: List[C], index: Int): Unit
+  def removePsK(txn: T, channels: List[C], patterns: List[P]): Unit
```

Comment:
Should that take an int, just like removeA does?

Reply:
Yeah, for consistency's sake it probably should.   I was using a different strategy, but better to keep it simple.

Reply:
Updated.